### DISCUSSION TITLE
Transaction log

### DISCUSSION THREAD

**Discussion by mdavid217** [eprbell/rp2#36] [Show and tell]
Did a little shopping the other day, and I wanted a way to be able to see a log of what I had done (in part to also ensure I recorded everything, and also I share some of my transactions with friends.) I wrote a quick plugin to generate a report that combines all transactions into a single sheet. It's somewhat ad-hoc and mostly suits my needs, but I cleaned it up a bit and threw it into a branch if you wanted to take a look. If you want a PR, I can do that, otherwise I have it for my own purposes.

Big ups to you for writing something flexible and extendable without a lot of effort. Thank you!

https://github.com/mdavid217/rp2/tree/transaction_log

**Comment by eprbell** [comment]
Thanks! This made me think: I do want people to experiment with the plugin architecture but we don't want every experimental plugin to be run by default. Perhaps we can have two classes of plugins: default and optional. The default ones would be ran by default, the optional ones would have to be selected by the user (your transaction list one could be one of the optional ones). Let me mull this over a bit more...

**Reply by mdavid217** [reply]
I think that's smart. It could get out of hand over time otherwise.

**Reply by eprbell** [reply]
In any case, while I think about this, please do open a PR: it's easier for me to review it.

**Reply by mdavid217** [reply]
TIL about running mypy on the individual file vs. running from project root... Sorry for the extra overhead processing as I sorted out those mypy tweaks. :)
